Update signal event handling.

jheer opened the pull request:
Changes:

**vega**
- Add `scatter-brush-filter` and `scatter-brush-panzoom` test specifications.

**vega-dataflow**
- Streamline signal event handler code.

**vega-view**
- Fix View event extension logic.

Close #1689.

arvind commented:
As I was able to address the cycle in the dataflow graph without the `rerank` property, my vote would be to not expose this type of low-level control in Vega unless/until we absolutely need it.

jheer commented:
Thanks @arvind, I've dropped the `rerank` option, and updated this PR to include only minor updates and some new selection-based test examples to help prevent future issues.
